src/main/java/io/cryostat/agent/ConfigModule.java

```diff
@@ -254,12 +268,77 @@ public static int provideCryostatAgentWebclientResponseTimeoutMs(Config config)
         return config.getValue(CRYOSTAT_AGENT_WEBCLIENT_RESPONSE_TIMEOUT_MS, int.class);
     }
 
+    @Provides
+    @Singleton
+    @Named(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PATH)
+    public static Optional<String> provideCryostatAgentWebclientTlsTruststorePath(Config config) {
+        return config.getOptionalValue(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PATH, String.class);
+    }
+
+    @Provides
+    @Singleton
+    @Named(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS_FILE)
+    public static Optional<ByteBuffer> provideCryostatAgentWebclientTlsTruststorePassFromFile(
+            Config config,
+            @Named(CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS_CHARSET) String passCharset) {
+        Optional<String> truststorePassFile =
+                config.getOptionalValue(
+                        CRYOSTAT_AGENT_WEBCLIENT_TLS_TRUSTSTORE_PASS_FILE, String.class);
+        if (truststorePassFile.isEmpty()) {
+            return Optional.empty();
+        }
+        try (FileInputStream passFile = new FileInputStream(truststorePassFile.get())) {
+            String pass = IOUtils.toString(passFile, Charset.forName(passCharset)).trim();
+            return Optional.ofNullable(new ByteBuffer(pass, passCharset));
```

Comment:
Is there some way to go directly from `FileInputStream` into `ByteBuffer`, without reading out the file contents with ex. `IOUtils.toString()`? This is still somewhat vulnerable to the problem I outlined before - this `String pass` gets allocated on the heap and contains the password, and when this variable drops out of scope it becomes eligible for garbage collection but will remain on the heap until that happens, which can take some time (or never if the host JVM happens to be using EpsilonGC). At least now it isn't held on the heap indefinitely, but it seems there could still be room to improve this by reading bytes directly from the input stream and copying them into the buffer's `byte[]`, so that they can be explicitly and deterministically cleared.

Reply:
ohh hmm I will take a look

Reply:
https://stackoverflow.com/questions/1264709/convert-inputstream-to-byte-array-in-java

I think a simple `passFile.readAllBytes()` or `passFile.toByteArray()` would do the trick here, plus a little refactoring of the `ByteBuffer`.